src/main/java/com/scalar/db/storage/cosmos/MutateStatementHandler.java

```diff
@@ -42,42 +50,71 @@ public MutateStatementHandler(CosmosClient client, TableMetadataManager metadata
       List<Record> results = execute(operation);
 
       return results;
-    } catch (NoMutationException e) {
-      throw e;
-    } catch (RuntimeException e) {
-      LOGGER.error(e.getMessage());
-      throw new ExecutionException(e.getMessage(), e);
+    } catch (CosmosException e) {
+      throwException(e);
     }
+
+    return Collections.emptyList();
+  }
+
+  protected void executeStoredProcedure(Mutation mutation) throws CosmosException {
+    executeStoredProcedure(Arrays.asList(mutation));
```

Comment:
I think it would be better if creating args is done here not to instantiate lots of ArrayList and copying objects for a single mutation by doing like the following.
What do you think ?

```
  protected void executeStoredProcedure(Mutation mutation) throws CosmosException {
    List<Object> args = new ArrayList<>();
    args.add(1);
    args.add(etMutationType(mutation).ordinal());
    args.add(makeRecord(mutation));
    args.add(makeConditionalQuery(mutation));
    
    executeStoredProcedure(mutation, args);
  }

  protected void executeStoredProcedure(List<? extends Mutation> mutations) throws CosmosException {
    List<Integer> types = new ArrayList<>();
    List<Record> records = new ArrayList<>();
    List<String> queries = new ArrayList<>();

    mutations.forEach(
        mutation -> {
          types.add(getMutationType(mutation).ordinal());
          records.add(makeRecord(mutation));
          queries.add(makeConditionalQuery(mutation));
        });
    List<Object> args = new ArrayList<>();
    args.add(mutations.size());
    args.addAll(types);
    args.addAll(records);
    args.addAll(queries);

    executeStoredProcedure(mutations.get(0), args);
  }

  private void executeStoredProcedure(Mutation mutation, List<Object> args) {
    CosmosStoredProcedureRequestOptions options =
        new CosmosStoredProcedureRequestOptions()
            .setPartitionKey(new PartitionKey(getConcatenatedPartitionKey(mutation));

    CosmosStoredProcedure storedProcedure =
        getContainer(mutation.getScripts().getStoredProcedure(MUTATION_STORED_PROCEDURE);
    CosmosStoredProcedureResponse response = storedProcedure.execute(args, options);
  }

Reply:
Thank you! I've modified that.